`test_replicated_users` failing tests when checked against flakes

### Company or project name

_No response_

### Describe the problem

When running flake tests on https://github.com/ClickHouse/ClickHouse/pull/81245

Failure report on pr:
https://s3.amazonaws.com/clickhouse-test-reports/PRs/81245/296215152e942ad68f1a452399faf2d24d3d1ae6//integration_tests_asan_flaky_check/integration_run_flaky0_0.log

I noticed failures for `test_replicated_users`. I initially thought it was my PR that caused this but it looks like this test has been failing for a long time when run with args `--count 3 --random`. Testing this on master reproduces the same failure.

Simple steps to reproduce locally:

```
# public repo on master branch

ubuntu@ip-10-1-1-1:~/work/ClickHouse/cmake-build-debug$ git branch
WARNING: terminal is not fully functional
Press RETURN to continue
* master
  pullout-replication-from-replicated-access


# print out last commit, head points to origin/master

ubuntu@ip-10-1-1-1:~/work/ClickHouse/cmake-build-debug$ git log --name-status HEAD^..HEAD
WARNING: terminal is not fully functional
Press RETURN to continue
commit fc0e372387eecaa577e5b1d3faee5498a50091b4 (HEAD -> master, origin/master, origin/cherrypick/25.4/80005, origin/cherrypick/25.3/80005, origin/cherrypick/24.8/80005, origin/HEAD)
Merge: ef4cce82a66 48506983df6
Author: Daniil Ivanik <61067749+divanik@users.noreply.github.com>
Date:   Wed Jun 4 12:26:07 2025 +0000

    Merge pull request #80005 from ClickHouse/divanik/disable_hive_partitioning_for_datalakes

    Disable hive partitioning for data lakes


# latest build

ubuntu@ip-10-1-1-1:~/work/ClickHouse/cmake-build-debug$ ninja clickhouse
[0/2] Re-checking globbed directories...
[0/7] cd /home/ubuntu/work/ClickHouse/rust/workspace/prql && /usr/bin/cmake -E env "CFLAGS= --gcc-toolchain=/home/...ickHouse/cmake-build-debug/contrib/corrosion-cmake/config.toml --offline --locked -- -Cdefault-linker-libraries=no
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.40s
[1/7] cd /home/ubuntu/work/ClickHouse/contrib/delta-kernel-rs/ffi && /usr/bin/cmake -E env "CFLAGS= --gcc-toolchai...ickHouse/cmake-build-debug/contrib/corrosion-cmake/config.toml --offline --locked -- -Cdefault-linker-libraries=no
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.31s
[3/7] cd /home/ubuntu/work/ClickHouse/rust/workspace/skim && /usr/bin/cmake -E env "CFLAGS= --gcc-toolchain=/home/...ickHouse/cmake-build-debug/contrib/corrosion-cmake/config.toml --offline --locked -- -Cdefault-linker-libraries=no
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.41s
[6/7] Running utility command for _cargo-build__ch_rust_skim_rust
ubuntu@ip-10-1-1-1:~/work/ClickHouse/cmake-build-debug$ ls -l programs/clickhouse
-rwxrwxr-x 1 ubuntu ubuntu 5287683008 Jun  4 18:36 programs/clickhouse




# running tests

ubuntu@ip-10-1-1-1:~/work/ClickHouse/tests/integration$ sudo ./runner --binary $HOME/work/ClickHouse/cmake-build-debug/programs/clickhouse --count 2 --random 'test_replicated_users' -- -ss

... (skipping test output and pasting summary instead)


=============================================================================================== 17 failed, 49 passed in 70.42s (0:01:10) ===============================================================================================
```

### Operating system

_No response_

### CMake version

_No response_

### Ninja version

_No response_

### Compiler name and version

_No response_

### Full cmake and/or ninja output with the error

```
======================================================================================================= short test summary info ========================================================================================================
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster[USER-1-2] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout:
FAILED test_replicated_users/test.py::test_create_replicated[USER-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster[ROW_POLICY-1-2] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout:
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster_ignore[ROW_POLICY-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster[QUOTA-1-2] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout:
FAILED test_replicated_users/test.py::test_create_replicated[ROW_POLICY-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_and_delete_replicated[QUOTA-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_rename_replicated[QUOTA-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_replicated[ROW_POLICY-2-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster[ROLE-2-2] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout:
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster[QUOTA-2-2] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout:
FAILED test_replicated_users/test.py::test_create_replicated[ROLE-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_and_delete_replicated[ROLE-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_and_delete_replicated[ROLE-2-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_create_replicated_on_cluster[SETTINGS_PROFILE-2-2] - helpers.client.QueryRuntimeException: Client expected to be failed but succeeded! stdout:
FAILED test_replicated_users/test.py::test_rename_replicated[SETTINGS_PROFILE-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
FAILED test_replicated_users/test.py::test_rename_replicated[ROLE-1-2] - helpers.client.QueryRuntimeException: Client failed! Return code: 237, stderr: Received exception from server (version 25.6.1):
=============================================================================================== 17 failed, 49 passed in 70.42s (0:01:10) ===========================================================================================
```

--- a/tests/integration/test_replicated_users/test.py
+++ b/tests/integration/test_replicated_users/test.py
@@ -1,5 +1,4 @@
 import inspect
-import time
 from dataclasses import dataclass
 from os import path as p
 
@@ -121,6 +120,20 @@ def test_create_replicated_on_cluster_ignore(started_cluster, entity):
 
     node1.query(f"DROP {entity.keyword} {entity.name} {entity.options}")
 
+    node1.replace_config(
+        "/etc/clickhouse-server/users.d/users.xml",
+        inspect.cleandoc(
+            f"""
+            <clickhouse>
+                <profiles>
+                    <default/>
+                </profiles>
+            </clickhouse>
+            """
+        ),
+    )
+    node1.query("SYSTEM RELOAD CONFIG")
+
 
 @pytest.mark.parametrize(
     "use_on_cluster",
@@ -147,14 +160,29 @@ def test_grant_revoke_replicated(started_cluster, use_on_cluster: bool):
     node1.query("SYSTEM RELOAD CONFIG")
     on_cluster = "ON CLUSTER default" if use_on_cluster else ""
 
-    node1.query(f"CREATE USER theuser {on_cluster}")
+    node1.query(f"CREATE USER theuser2 {on_cluster}")
 
-    assert node1.query(f"GRANT {on_cluster} SELECT ON *.* to theuser") == ""
+    assert node1.query(f"GRANT {on_cluster} SELECT ON *.* to theuser2") == ""
 
-    assert node2.query(f"SHOW GRANTS FOR theuser") == "GRANT SELECT ON *.* TO theuser\n"
+    assert node2.query(f"SHOW GRANTS FOR theuser2") == "GRANT SELECT ON *.* TO theuser2\n"
 
-    assert node1.query(f"REVOKE {on_cluster} SELECT ON *.* from theuser") == ""
-    node1.query(f"DROP USER theuser {on_cluster}")
+    assert node1.query(f"REVOKE {on_cluster} SELECT ON *.* from theuser2") == ""
+    node1.query(f"DROP USER theuser2 {on_cluster}")
+
+    node1.replace_config(
+        "/etc/clickhouse-server/users.d/users.xml",
+        inspect.cleandoc(
+            f"""
+            <clickhouse>
+                <profiles>
+                    <default/>
+                </profiles>
+            </clickhouse>
+            """
+        ),
+    )
+    node1.query("SYSTEM RELOAD CONFIG")
+
 
 
 @pytest.mark.parametrize("entity", entities, ids=get_entity_id)

